### ISSUE TITLE
Email notification with team as recipient is not sent

### ISSUE THREAD

**Issue by tomkuipers** [DependencyTrack/dependency-track#4302]
### Current Behavior

Email notification with team as recipient is not sent.

Setup a email notification alert and set a team as recipient. Select group 'new_vulnerability' under portfolio and press button 'perform test'. No email is send.

dependency-track-api log entry:

2024-10-24 12:21:21,758 WARN [SendMailPublisher] No destination(s) provided; Skipping notification (PublishContext{notificationGroup=NEW_VULNERABILITY, notificationLevel=WARNING, notificationScope=PORTFOLIO, notificationTimestamp=2024-10-24T12:21:21.757898137Z, notificationSubjects={component=Component[uuid=94f87321-a5d1-4c2f-b2fe-95165debebc6, group=null, name=componentName, version=componentVersion], projects=[Project[uuid=c9c9539a-e381-4b36-ac52-6a7ab83b2c95, name=projectName, version=projectVersion]], vulnerability=Vulnerability[id=INT-001, source=INTERNAL]}}) [requestId=c4ba9542-885d-4a60-a5fd-cca446b1b426]

### Steps to Reproduce

1. Administration > Notifications > Alerts
2. Select team with email as publisher
3. Verify Publisher class is `org.dependencytrack.notification.publisher.SendMailPublisher`
4. Verify that destination is empty
5. Select team as recipient contains a team
6. Under Portfolio check group 'NEW_VULNERABILITY'
7. Click 'Perform test'
8. Inspect dependency-check-api logs

### Expected Behavior

An OIDC group contains members,  this group is mapped to a team with same name. When I select the team as recipient in a email notification and perform a test, I expect all members of the OIDC group to receive an email.

### Dependency-Track Version

4.12.0

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Browser

Google Chrome

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [x] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by 2000rosser** [comment]
You would need to set the destination as your SMTP server for the emails to be sent

**Comment by 2000rosser** [comment]
Do you have the SMTP server configured in the email configuration section?

**Comment by 2000rosser** [comment]
I dont think the perform test is configured to use the destination there. I'll look into this issue.

**Comment by 2000rosser** [comment]
@tomkuipers is there a reason you aren't adding the SMTP server to the destination?

**Comment by tomkuipers** [comment]
> Do you have the SMTP server configured in the email configuration section?

Yes, I have and receive the test email. Also admin is receiving notifcations about occasional failures of bom parsing, so email notification is working.

**Comment by patrickfil** [comment]
Hey, we are currently facing the same issue. As @babinskiy mentioned an error is thrown when the destination is not set, even if a team is set as recipient. And if the Destination and a team is defined as recipient, the mail is only sent to the Destination. The api-server log is empty even if the "Log successful publish" is enabled.

The SMTP Server is also configured correct, as i receive the Test-Mail on the Mail-Address that is set on "Destination".

**Dependency-Track Version**
4.13.6

**Dependency-Track Distribution**
Helm-Chart Version 0.42.0

**Container Image**
dependencytrack/apiserver:4.13.6

**Database Server**
PostgreSQL

**Database Server Version**
17

**Checking logs:**
`kubectl logs dtrack-dependency-track-api-server-0 --since 5m`
